hypothesis-python/tests/array_api/test_arrays.py

```diff
@@ -350,7 +351,9 @@ def test_floats_can_be_constrained_excluding_endpoints(x):
 @given(
     xps.arrays(
         dtype=xp.float32,
-        elements=st.floats(allow_nan=False, width=32),
+        elements=st.floats(
+            allow_nan=False, allow_subnormal=not WIDTHS_FTZ[32], width=32
+        ),
```

Comment:
```suggestion
        elements={"allow_nan": False},
```

`elements` can be a mapping of keyword arguments to `from_dtype()`, which would be clearer here and in the cases above - `allow_subnormal` can then be inferred.

And it turns out that our type annotation for this is wrong, oops 😅

Reply:
Good catch! Yeah I had just ran regex to fix these, but is a good oppurtunity to replace `st.floats()` with mappings (unless we're specifically testing passing an elements strategy ofc). This is a vestige from the sibling `test_gen_data` tests, where I suspect these tests were written before you supported mappings.

Reply:
Relying on mappings now, and fixed the type hints for `xps.arrays`/`_arrays`.